### DISCUSSION TITLE
Scaling with Workspace plot methods

### DISCUSSION THREAD

**Discussion by ksebby** [whitews/FlowKit#211] [General]
Let me echo others by first saying thank you for developing this package. It is very well thought out and gets me out of R, which is not my favorite place to work... My use case is to generate custom reports after gating has been done manually in FlowJo. I would like the plots generated with FlowKit to accurately represent plots as seen in FlowJo. However, when I use workspace plotting methods--wsp.plot_gate and wsp.plot_scatter--the axes are always scaled from 0 to 1 and I don't see any additional arguments to control this. I get the same scaling as in FlowJo if I plot from the Sample class: sample.plot_scatter(). Is there a way to preserve the scaling when plotting from the Workspace class? I'd like to have the plots with the gates as wsp.plot_gate() does. Thanks!
Screenshots: 
1. wsp.plot_gate() showing axes scaled 0-1.
![image](https://github.com/user-attachments/assets/2928e630-4a60-47e6-a912-f955ebf24665)
2. wsp.plot_scatter() showing axes scaled 0-1.
![image](https://github.com/user-attachments/assets/ebc4dff4-feae-4d90-a5e7-1d97850d8015)
3. sample.plot_scatter() showing original scaling (scattering channels have no compensated or transformed events)
![image](https://github.com/user-attachments/assets/5a3e1c4d-269c-4567-87d7-2ff96108d46f)
4. Flowjo plot showing original scaling.
<img width="691" alt="image" src="https://github.com/user-attachments/assets/3b0772f6-348e-4654-b386-32efc958605e">

**Comment by whitews** [comment]
Hi Karl,

Thanks for the kind words and I'm glad to hear you find the library useful. And I sense we have similar feelings on R (no offense to the R folks!).

You are correct, FlowJo and FlowKit handle plotting differently regarding transformed data. This question has come up before and you can read the detailed explanation [here](https://github.com/whitews/FlowKit/issues/195#issuecomment-2086163780). The difference you are seeing in the 2 FlowKit plots (Sample vs Workspace classes) is due to FlowJo using linear transforms on scatter channels by default, which are applied to those plotted gates. With linear transforms, the plotted data looks identical but the axis values change.

There may be a way to re-define the axis labels in the returned Bokeh figure instance but I've never tried it so not 100% sure. If you go that route and find a solution let me know, I'd consider adding a keyword argument as an option.

Hope this answers your question, sorry I don't have an easy answer.

Best,
Scott

**Comment by ksebby** [comment]
Great. Thank you for the quick response. I missed the issue that you linked to. I always forget to search for closed issues. I'll have to read things through another time to make sure I understand it well. I'm new to Bokeh, so I'll have to get up to speed with it and see if I have any ideas. It creates beautiful plots. It would be good to have more control over the plots in general--fonts, labels, etc. Maybe the Workspace/Sample plotting methods could take arbitrary kwargs that are passed to the underlying plotting functions. I'll take a look. Thanks again.
